pytensor/tensor/subtensor.py

```diff
@@ -728,26 +738,39 @@ def make_node(self, x, *inputs):
         ) + [slice(None, None, None)] * (x.type.ndim - len(idx_list))
 
         out_shape = []
-        for i, (p, s) in enumerate(zip(padded, x.type.shape)):
-            if isinstance(p, slice):
-                if s == 1:
-                    start = p.start
-                    try:
+        for i, (the_slice, length) in enumerate(zip(padded, x.type.shape)):
+            if isinstance(the_slice, slice):
+                if length is None:
+                    out_shape.append(None)
+                    continue
+
+                start = the_slice.start
+                stop = the_slice.stop
+                step = the_slice.step
+
+                is_slice_const = True
+                try:
+                    if start is not None:
                         start = get_underlying_scalar_constant_value(start)
-                    except NotScalarConstantError:
-                        pass
-                    if start is None or start == 0:
-                        start = p.start
-                        if start is None:
-                            start = 0
-                        if p.stop is None or (
-                            isinstance(p.stop, (int, np.integer, np.ndarray))
-                            and p.stop > start
-                        ):
-                            out_shape.append(1)
-                            continue
-
-                out_shape.append(None)
+                except NotScalarConstantError:
```

Comment:
Nitpick can we do the 3 in one try/except block?

Reply:
done

Reply:
Didn't work out after all, we have to accept None values, and that raises an exception.

Reply:
Np

Reply:
There's also `get_scalar_constant` that may not raise with None. Not sure now